harness/determined/pytorch/_pytorch_context.py

```diff
@@ -57,4 +103,344 @@ def get_lr_scheduler(self) -> Optional[pytorch.LRScheduler]:
+
+        The LR scheduler must use an optimizer initialized by :meth:`Optimizer` because
+        ``torch.optim.LRScheduler`` directly access the optimizer parameter groups,
+        which are initialized in :meth:`Optimizer`
+        """
+
+        check.is_in(
+            lr_scheduler.optimizer,  # type: ignore
+            self.optimizers,
+            "Must use an optimizer that is returned by Optimizer()",
+        )
+        wrapped = pytorch.LRScheduler(lr_scheduler, step_mode)
+        self.lr_schedulers.append(wrapped)
+        return wrapped
+
+    def _filter_named_parameters(self, optimizer: torch.optim.Optimizer) -> List:  # type: ignore
+        """_filter_named_parameters filters the named parameters of a specified optimizer out
+        of all the named parameters from a specified model. We need this function because
+        a ``torch.optim.Optimizer`` doesn't store parameter names and we need the names of
+        the parameters when mapping parameters to each ``horovod.DistributedOptimizer``.
+        """
+        opt_params = {p for group in optimizer.param_groups for p in group.get("params", [])}
+        return [(name, p) for name, p in self._main_model.named_parameters() if p in opt_params]
+
+    def _init_device(self) -> None:
+        self.n_gpus = len(self.env.container_gpus)
+        if self.hvd_config.use:
+            check.gt(self.n_gpus, 0)
+            # We launch a horovod process per GPU. Each process
+            # needs to bind to a unique GPU.
+            self.device = torch.device(hvd.local_rank())
+            torch.cuda.set_device(self.device)
+        elif self.n_gpus > 0:
+            self.device = torch.device("cuda", 0)
+        else:
+            self.device = torch.device("cpu")
+        check.is_not_none(self.device)
+
+    def to_device(self, data: pytorch._Data) -> pytorch.TorchData:
+        return pytorch.to_device(
+            data, self.device, self.warning_logged[_WarningLogs.FAILED_MOVING_TO_DEVICE]
+        )
+
+    def configure_apex_amp(
+        self,
+        models: Union[torch.nn.Module, List[torch.nn.Module]],
+        optimizers: Union[torch.optim.Optimizer, List[torch.optim.Optimizer]],  # type: ignore
+        enabled: Optional[bool] = True,
+        opt_level: Optional[str] = "O1",
+        cast_model_type: Optional[torch.dtype] = None,
+        patch_torch_functions: Optional[bool] = None,
+        keep_batchnorm_fp32: Optional[Union[bool, str]] = None,
+        master_weights: Optional[bool] = None,
+        loss_scale: Optional[Union[float, str]] = None,
+        cast_model_outputs: Optional[torch.dtype] = None,
+        num_losses: Optional[int] = 1,
+        verbosity: Optional[int] = 1,
+        min_loss_scale: Optional[float] = None,
+        max_loss_scale: Optional[float] = 2.0 ** 24,
+    ) -> Tuple:
+        """
+        Configure automatic mixed precision for your model and optimizers. Note that Determined
+        integrates apex.amp internally for AMP features.
```

Comment:
What do you think about replacing:

"Note that Determined integrates apex.amp internally for AMP features"

with:

"Note that details for apex.amp are handled automatically within Determined after this call"